src/java.base/share/classes/java/lang/classfile/Signature.java

```diff
@@ -222,12 +243,20 @@ public static ClassTypeSig of(String className, TypeArg... typeArgs) {
          * {@return a class type signature}
          *
          * @param outerType signature of the outer type, may be {@code null}
+         *        to indicate this is a top-level class or interface
```

Comment:
Actually here is the ambiguity. Null outer type is more than frequently used also for inner classes.
Signatures of inner classes are mainly encoded without the outer type and with the $-notation of the outer$inner as the class name.

Reply:
Clarified by introducing examples of signature strings and on the className/outerType.